src/contrib/mlir/core/ngraph_dialect/fused_ops.td

```diff
@@ -32,14 +33,899 @@ def NGSqueezeOp :
+
+  let builders = [
+    OpBuilder<
+      "Builder *builder, OperationState &tblgen_state, Type res,"
+      "Value *X, Value* W, Value* R, Value* H_t,"
+      "Attribute hiddenSize, ArrayAttr activations,"
+      "ArrayAttr activationAlpha, ArrayAttr activationBeta,"
+      "Attribute clip, Attribute linearBeforeReset", [{
+        tblgen_state.addOperands({X, W, R, H_t});
+        tblgen_state.addAttribute("hiddenSize", hiddenSize);
+        tblgen_state.addAttribute("activations", activations);
+        tblgen_state.addAttribute("activationAlpha", activationAlpha);
+        tblgen_state.addAttribute("activationBeta", activationBeta);
+        tblgen_state.addAttribute("linearBeforeReset", linearBeforeReset);
+        tblgen_state.addTypes(res);
+      }]>,
+
+    OpBuilder<
+      "Builder *builder, OperationState &tblgen_state, Type res,"
+      "Value *X, Value* W, Value* R, Value* H_t,"
+      "Attribute hiddenSize",
+      [{
+        tblgen_state.addOperands({X, W, R, H_t});
+        tblgen_state.addAttribute("hiddenSize", hiddenSize);
+        tblgen_state.addTypes(res);
+      }]>
+    ];
+
+    let extraClassDeclaration = [{
+      void setHiddenSize  (const Attribute& attr)       { this->setAttr("hiddenSize", attr); }
+      void setActivations (const ArrayAttr& attr)       { this->setAttr("activation", attr); }
+      void setActivationsAlpha (const ArrayAttr& attr)  { this->setAttr("activatiAlpha", attr); }
+      void setActivationsBeta (const ArrayAttr& attr)   { this->setAttr("activatiBeta", attr); }
+      void setLinearBeforeReset(const Attribute& attr)  { this->setAttr("linearBeforeReset", attr); }
+
+    }];
+}
+
+// LayerNorm Op
+
+// Op produces 3 results when keepStats is true
+// Op is defined with 3 results, 2 of which are invalid/dead if keepStats is false
+
+def NGLayerNormOp : 
+    NG_Op<"layernorm", [NoSideEffect, DeclareOpInterfaceMethods<FusedOp>]>,
+    Results<(outs NG_TensorType:$res, NG_TensorType:$mean, NG_TensorType:$var)>,
+    Arguments<(ins NG_TensorType:$data, NG_TensorType:$scale, NG_TensorType:$bias,
+               DefaultValuedAttr<BoolAttr, "true">:$keepStats,
+               DefaultValuedAttr<I64Attr, "1">    :$beginNormAxis,
+               DefaultValuedAttr<F32Attr, "1e-5"> :$epsilon)>
+{
+  let summary = "LayerNorm Op";
+  let description = "Constructs an LayerNorm operation.";
+    
+  let builders = [
+    OpBuilder<
+    "Builder *builder, OperationState &tblgen_state, ArrayRef<Type> res,"
+    "Value *data, Attribute keepStats, Attribute beginNormAxis, Attribute epsilon", [{
+       tblgen_state.addOperands(data);
+       tblgen_state.addAttribute("keepStats", keepStats);
+       tblgen_state.addAttribute("beginNormAxis", beginNormAxis);
+       tblgen_state.addAttribute("epsilon", epsilon);
+       tblgen_state.addTypes(res);
+    }]>
+  ];
+
+  let extraClassDeclaration = [{
+    void setKeepstats(const Attribute& attr)      { this->setAttr("keepStats", attr);    }
```

Comment:
just checking: would you need the getter functions too? ngraph Fusedop defines getters.